File: TypeScript/class.ts
class Person {
    name: string;
    private age: number;
    
    constructor(name: string, age: number) {
        this.name = name;
        this.age = age;
    }

    info() {
        return this.name + " is " + this.age + " years old.";
    }
}

let person = new Person("John", 30);
console.log(person.name);
// console.log(person.age); // illegal
console.log(person.info());

File: Dart/class.dart
class Person {
    String name;
    int age;

    /*Person(String _name, int _age) {
        name = _name;
        age = _age;
    }*/

    /*Person(String name, int age) {
        this.name = name;
        this.age = age;
    }*/

    Person(String this.name, int this.age);

    info() {
        return "$name is $age.";
    }
}

main() {
    var p = Person("John", 20);
    print(p.info());
}